libethereum/BlockDetails.h

```diff
@@ -17,27 +17,39 @@ namespace eth
 
 // TODO: OPTIMISE: constructors take bytes, RLP used only in necessary classes.
 
-static const unsigned c_bloomIndexSize = 16;
-static const unsigned c_bloomIndexLevels = 2;
+constexpr unsigned c_bloomIndexSize = 16;
+constexpr unsigned c_bloomIndexLevels = 2;
 
-static const unsigned c_invalidNumber = (unsigned)-1;
+constexpr unsigned c_invalidNumber = (unsigned)-1;
 
 struct BlockDetails
 {
 	BlockDetails(): number(c_invalidNumber), totalDifficulty(Invalid256) {}
-	BlockDetails(unsigned _n, u256 _tD, h256 _p, h256s _c): number(_n), totalDifficulty(_tD), parent(_p), children(_c) {}
-	BlockDetails(RLP const& _r);
+    BlockDetails(unsigned _number, u256 _totalDifficulty, h256 _parentHash, h256s _childHashes,
+        unsigned _blockSizeBytes)
+      : number{_number},
+        totalDifficulty{_totalDifficulty},
+        parentHash{_parentHash},
+        childHashes{_childHashes},
+        blockSizeBytes{_blockSizeBytes}
+    {}
+    BlockDetails(RLP const& _r);
 	bytes rlp() const;
 
 	bool isNull() const { return number == c_invalidNumber; }
 	explicit operator bool() const { return !isNull(); }
 
 	unsigned number = c_invalidNumber;
 	u256 totalDifficulty = Invalid256;
-	h256 parent;
-	h256s children;
+	h256 parentHash;
+	h256s childHashes;
 
-	mutable unsigned size;
+    // Size of the BlockDetails RLP (in bytes). Used for computing blockchain memory usage
+    // statistics. Field name must be 'size' as BlockChain::getHashSize depends on this
+    mutable unsigned size;
```

Comment:
Is it still useful for anything?

Reply:
@gumb0 Not sure, will see if it can be removed

Reply:
It's used to compute BlockChain memory usage stats which can be used to garbage collect data from the BlockChain cache:

Memory usage stats computation: https://github.com/ethereum/aleth/blob/43780e1cfaad9d887b5cd730a5df6abcea81f302/libethereum/BlockChain.cpp#L1184-L1205

Garbage collection: https://github.com/ethereum/aleth/blob/43780e1cfaad9d887b5cd730a5df6abcea81f302/libethereum/BlockChain.cpp#L1207-L1271

Reply:
Then I think you should still set it in ` BlockDetails::rlp()` function

Reply:
@gumb0 Oops yes I definitely should 😀